### PR TITLE
feat: auto focus search input

### LINKED ISSUE #81: Add `/` shortcut for search
It would be nice to type `/` and add the focus to the search input, so I won't have to leave my keyboard to search for a module.

Of course, we should be able to type `/` into the search bar.

### PR DESCRIPTION
**Pull request by Geminii**
Resolves #81 

- [x] add "/" shortcut to auto focus search input
- [x] add visual to improve the UX to focus the input
Capture d’écran 2021-01-14 à 18 29 04

- [x] auto focus search input on desktop after loading

### CODE REVIEW

**Comment by Geminii** [comment]
> It would be nice indeed @Geminii :)

I admit that I am not inspired by this wording 😅 because of the characters length. 
Current sentence : 
> Search a module (name, category, username, etc.)

Propositions : 
> Use "/" to search a module (name, category, username, etc.)
> Search ("/" to focus) a module (name, category, username, etc.) // A little bit too long

@Atinux your suggestions are welcome ❤️

**Comment by Atinux** [comment]
I would simply add a small box with / floating at the right of the search

File: pages/index.vue [hunk 1]
```diff
@@ -159,6 +173,7 @@ import categories from '~/categories'
 import CardModule from '~/components/CardModule.vue'
 import Observer from '~/components/Observer.vue'
 import { numberFormatter } from '~/utils/format.ts'
+import { detectMobile } from '~/utils/detectUserAgent.ts'
```
Comment (Atinux) [inline pages/index.vue:176]:
```suggestion
import { isMobile } from '~/utils/detectUserAgent.ts'
```

File: pages/index.vue [hunk 2]
```diff
@@ -333,6 +349,19 @@ export default {
...
+    // Add `/` shortcut for search input only if not already focused
+    document.addEventListener('keypress', e => {
+      if (e.keyCode === 47 && !this.searchFocus) {
+        e.preventDefault()
+        this.focusSearchInput()
+      }
+    })
+
+    // In case of desktop, auto focus the search input
+    if (!detectMobile()) {
```
Comment (Atinux) [inline pages/index.vue:362]:
```suggestion
    if (!isMobile()) {
```

File: utils/detectUserAgent.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+export function detectMobile(): boolean {
```
Comment (Atinux) [inline utils/detectUserAgent.ts:1]:
```suggestion
export function isMobile(): boolean {
```

File: utils/detectUserAgent.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  const toMatch = [
+      /Android/i,
+      /webOS/i,
+      /iPhone/i,
+      /iPad/i,
+      /iPod/i,
+      /BlackBerry/i,
+      /Windows Phone/i
+  ];
```
Comment (Atinux) [inline utils/detectUserAgent.ts:10]:
```suggestion
  ]
```

File: utils/detectUserAgent.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+  return toMatch.some((toMatchItem: RegExp) => {
+      return navigator.userAgent.match(toMatchItem);
+  });
```
Comment (Atinux) [inline utils/detectUserAgent.ts:14]:
```suggestion
  })
```

File: pages/index.vue [hunk 6]
```diff
@@ -333,6 +349,19 @@ export default {
     if (orderBy) {
       this.orderBy = orderBy
     }
+    
+    // Add `/` shortcut for search input only if not already focused
+    document.addEventListener('keypress', e => {
```
Comment (Atinux) [inline pages/index.vue:354]: We need to store this eventListerner to remove it on `beforeDestroy()` hook

**Review by Atinux — CHANGES_REQUESTED** [review]

File: pages/index.vue (hunk 6 above)
Comment (Geminii) [reply pages/index.vue]: I added hook, a good point thanks 🙏

**Pull request merged**